lsp-mode.el

```diff
@@ -5224,10 +5279,27 @@ It will show up only if current point has signature help."
       (funcall action-handler action)
     (lsp--send-execute-command command arguments?)))
 
+(defun lsp--select-action-enabled-or-die (actions)
+  "Prompt the user to select an action from ACTIONS.
+Even disabled code actions are shown. If such an action is
+chosen, `user-error'."
+  (let ((action (lsp--select-action actions)))
+    (-when-let ((&CodeAction :disabled?) action)
+      (user-error "%s is disabled: %s" (lsp:code-action-title action)
+                  (lsp:code-action-disabled-reason disabled?)))
```

Comment:
this seems like should be part of `lsp-execute-code-action`. E. g. if lsp-ui calls the method with disabled action it should get that error.

Reply:
Fixed.

Reply:
Note: this is now an `error` instead of a `user-error` (reasoning in the commit message).